packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -47,13 +49,21 @@ export class FabricBase extends React.Component<
     const { as: Root = 'div', theme, dir } = this.props;
     const classNames = this._getClassNames();
     const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties, ['dir']);
-    const componentDir = getDir(theme, dir);
-    const parentDir = getDir(theme);
+    const getDir = getAllDir(theme, dir);
+    // If Fabric dir !== theme.RTL
+    // Or If theme.RTL !== page RTL
+    // Then we need to set dir on the Fabric component
+    const setComponentDir = getDir.component !== getDir.parent || getDir.component !== getDir.page;
```

Comment:
if getDir is being replaced by getAllDir above how are these calls to getDir still valid?

Reply:
i just reorganized the logic.